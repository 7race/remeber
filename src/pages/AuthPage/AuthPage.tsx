import { Button } from '../../atoms/Button/Button';

export const AuthPage = () => (
  <Button variant='success' outlined>
    Войти
  </Button>
);
