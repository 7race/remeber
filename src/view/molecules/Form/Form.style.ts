import styled from 'styled-components';

export const Form = styled.form`
  border-radius: 16px;
  display: flex;
  flex-direction: column;
  & p {
    text-align: center;
    margin-bottom: 46px;
  }
  & input {
    width: 100%;
    background-color: white;
  }
  & > input {
    margin-bottom: 40px;
  }
  & div:first-of-type {
    width: 100%;
    margin-bottom: 60px;
  }
  & button {
    width: 100%;
  }

  @media ${({ theme }) => theme.media.notebook} {
    background-color: ${({ theme }) => theme.colors.white};
    padding: 60px 50px;
    width: 512px;
    height: 550px;
    & input {
      background-color: ${({ theme }) => theme.colors.lightGrey};
    }
    & div:first-of-type {
      margin-bottom: 80px;
    }
  }
`;
