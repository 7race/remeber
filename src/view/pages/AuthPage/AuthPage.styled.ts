import styled, { createGlobalStyle } from 'styled-components';

export const GlobalStyle = createGlobalStyle`
  body {
    background: ${({ theme }) => theme.colors.lightGrey};
  }
`;

export const AuthPage = styled.div`
  display: grid;
  height: 100vh;
  grid-template-rows: 20vh minmax(min-content, max-content) 1fr;
  & > form {
    grid-area: form;
  }
  grid-template-areas:
    '.'
    'form'
    '.';
  padding: 20px;

  @media ${({ theme }) => theme.media.tablet} {
    grid-template-rows: 1fr minmax(min-content, max-content) 1fr;
    grid-template-columns: 1fr 400px 1fr;
    grid-template-areas:
      '. . .'
      '. form .'
      '. . .';
  }
`;
