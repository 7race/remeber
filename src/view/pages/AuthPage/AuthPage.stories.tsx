import { ComponentStory, ComponentMeta } from '@storybook/react';
import { AuthPage } from './AuthPage';

export default {
  title: 'AuthPage',
  component: AuthPage,
} as ComponentMeta<typeof AuthPage>;

const Template: ComponentStory<typeof AuthPage> = () => <AuthPage />;

export const Default = Template.bind({});
