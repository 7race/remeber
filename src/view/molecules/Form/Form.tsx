import { MouseEvent } from 'react';
import { P } from '@atoms/P';
import { Input } from '@atoms/Input';
import { Button } from '@atoms/Button';
import * as S from './Form.style';

export const Form = () => {
  const submitClick = (e: MouseEvent) => {
    e.preventDefault();
  };

  return (
    <S.Form aria-label='form'>
      <P type='h1'>Вход</P>
      <Input placeholder='Логин' variant='grey' />
      <Input placeholder='Пароль' type='password' variant='grey' />
      <Button variant='submit' onClick={submitClick}>
        Войти
      </Button>
    </S.Form>
  );
};
