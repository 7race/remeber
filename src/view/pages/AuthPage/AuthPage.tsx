import { Form } from '@molecules/Form';
import * as S from './AuthPage.styled';

export const AuthPage = () => (
  <S.AuthPage>
    <S.GlobalStyle />
    <Form />
  </S.AuthPage>
);
