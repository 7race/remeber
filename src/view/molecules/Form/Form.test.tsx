import { render, screen } from '@tests-utils';
import { AuthPage } from '@pages/AuthPage';
import { Form } from './Form';

describe('Form in auth page', () => {
  it('should render', () => {
    render(<AuthPage />);
    const form = screen.getByRole('form');
    expect(form).toBeInTheDocument();
  });
  it('should have correct styles', () => {
    const { container } = render(<Form />);
    expect(container).toMatchSnapshot();
  });
});
