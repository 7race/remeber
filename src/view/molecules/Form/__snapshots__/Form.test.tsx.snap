// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form in auth page should have correct styles 1`] = `
<div>
  <form
    aria-label="form"
    class="sc-eCYdqJ fIvkZv"
  >
    <p
      class="sc-bczRLJ Gujvk"
      type="h1"
    >
      Вход
    </p>
    <input
      class="sc-dkzDqf dVlRoD"
      placeholder="Логин"
    />
    <div
      class="sc-gsnTZi dGUFFC"
    >
      <input
        class="sc-dkzDqf dVlRoD"
        placeholder="Пароль"
        type="password"
      />
      <img
        alt="closed eye"
        src="eye-slash.svg"
      />
    </div>
    <div
      class="react-ripples"
      style="position: relative; display: inline-flex; overflow: hidden;"
    >
      <button
        class="sc-hKMtZM blpuEy"
      >
        Войти
      </button>
      <s
        style="position: absolute; border-radius: 50%; opacity: 0; width: 35px; height: 35px; transform: translate(-50%, -50%); pointer-events: none;"
      />
    </div>
  </form>
</div>
`;
